activerecord/lib/active_record/sanitization.rb

```diff
@@ -165,7 +165,9 @@ def replace_named_bind_variables(statement, bind_vars)
 
         def quote_bound_value(value, c = connection)
           if value.respond_to?(:map) && !value.acts_like?(:string)
-            if value.respond_to?(:empty?) && value.empty?
+            is_empty = (value.respond_to?(:empty?) && value.empty?) ||
+              (value.is_a?(Range) && value.none?)
```

Comment:
Instead of adding an extra special-case for Range, how about we move the `empty?` check to after the `map` -- at that point we know it's an array, so it's safe to call.

That way we'll fix the problem for all enumerables, and not just go from (ones that support `empty?`) to ((ones that support `empty?`) + `Range`).

```ruby
quoted = value.map { |v| c.quote(v) }
if quoted.empty?
  c.quote(nil)
else
  quoted.join(",")
end
```

Reply:
Yep, that's a better approach. I'll make your proposed change and rebase.